TowerForge/TowerForge/GameModule/Systems/LevelSystems/HomeSystem.swift

```diff
@@ -52,16 +54,48 @@ class HomeSystem: TFSystem {
         }
 
         // Check if they have enough points to spawn
-        for playerHomeComponent in playerHomeComponentArr {
-            guard playerHomeComponent.points >= type.cost else {
-                return
-            }
-            playerHomeComponent.decreasePoints(type.cost)
+        guard !playerHomeComponentArr.contains(where: { $0.points < type.cost }) else {
+            return
         }
 
-        let snapPosition = CGPoint(x: position.x, y: gridDelegate.snapYPosition(yPosition: position.y))
+        // Reduce points
+        playerHomeComponentArr.forEach({ $0.decreasePoints(type.cost) })
+
+        let snapPosition = getSnapPosition(at: position, for: player, with: type)
         let spawnEvent = SpawnEvent(ofType: type, timestamp: CACurrentMediaTime(),
                                     position: snapPosition, player: player)
         eventManager.add(spawnEvent)
     }
+
+    private func getSnapPosition<T: TFEntity & PlayerSpawnable>(at requestedPosition: CGPoint,
```

Comment:
should the responsibility of snapping to grid not be left to HomeSystem?

Reply:
Most of the playable logic is handled by the HomeSystem and only the snapping to the center of each grid square is handled by the gridDelegate which i feel is fine to leave outside the homeSystem.